### PR TITLE
Feature/oauthdemo

### PR DESCRIPTION
**Pull request by coolbnjmn**
https://nomtek.atlassian.net/browse/CRI-992

Add APIClient demo + documentation

### CODE REVIEW

File: CreatubblesAPIClientDemo/DemoAPIClient.swift [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+//
+//  Created by Benjamin Hendricks on 01/05/2017.
+//  Copyright © 2017 Nomtek. All rights reserved.
+//
+
+import CreatubblesAPIClient
+import UIKit
+
+struct OAuthExampleConstants {
+    static let appId = "acbf03d5084f0d97159587e819e27b688a19b313b93db09c19f72ee1577b244d"
```
Comment (mmiedlarz) [inline CreatubblesAPIClientDemo/DemoAPIClient.swift]: I really prefer to not put any keys in private repo. Could you force push clean version of this?

File: CreatubblesAPIClientDemo/DemoAPIClient.swift [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+            }
+            print("WARNING: Public authentication was not set properly:\(error)")
+        }
+    }
+
+    fileprivate static func prepareClient() -> CreatubblesAPIClient.APIClient {
+        return CreatubblesAPIClient.APIClient(settings: OAuthExampleConstants.settings)
+    }
+
+    //        APIClientAccessSingleton.sharedInstance.client.delegate = self
```
Comment (mmiedlarz) [inline CreatubblesAPIClientDemo/DemoAPIClient.swift]: commented code

File: CreatubblesAPIClientDemo/DemoWelcomeViewController.swift [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+}
+
+extension DemoWelcomeViewController: SFSafariViewControllerDelegate {
+    public func handleOpenUrl(url: URL) {
+        let codeKey = "code="
+        let queryParams = url.query?.components(separatedBy: "&")
+        let codeParam = queryParams?.filter {
+            element in
+            // 5 becaues "code=" has 5 characters.
+            return String(element.characters.prefix(5)) == codeKey
```
Comment (mmiedlarz) [inline CreatubblesAPIClientDemo/DemoWelcomeViewController.swift]:
```
return String(element.characters.prefix(codeKey.characters.count)
```

File: Documentation/OauthDemo.md [hunk 4]
```diff
@@ -0,0 +1,47 @@
+# CreatubblesAPIClient Demo App
```
Comment (mmiedlarz) [inline Documentation/OauthDemo.md:1]: Is it linked anywhere? I don't see changes in `Readme.md`

**Review by mmiedlarz — CHANGES_REQUESTED** [review]
Code:
- There are shared keys and passwords added to public repo.
- Some minor things

Func: 👍

File: Documentation/OauthDemo.md (hunk 4 above)
Comment (coolbnjmn) [reply Documentation/OauthDemo.md:1]: Added a link in `README.md` :D

**Review by mmiedlarz — APPROVED** [review]
Minor change mentioned on slack

**Pull request merged**